Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1722,7 +1720,12 @@ void WebPageProxy::maybeInitializeSandboxExtensionHandle(WebProcessProxy& proces
 #endif
     SandboxExtension::Handle sandboxExtensionHandle;
     if (!resourceDirectoryURL.isEmpty()) {
-        if (checkAssumedReadAccessToResourceURL && process.hasAssumedReadAccessToURL(resourceDirectoryURL))
+#if PLATFORM(COCOA)
+        bool hasSandboxAccess = !sandbox_check(process.processID(), "file-read-data", static_cast<enum sandbox_filter_type>(SANDBOX_FILTER_PATH | SANDBOX_CHECK_NO_REPORT), resourceDirectoryURL.fileSystemPath().utf8().data());
+#else
+        bool hasSandboxAccess = true;
+#endif
+        if (checkAssumedReadAccessToResourceURL && process.hasAssumedReadAccessToURL(resourceDirectoryURL) && hasSandboxAccess)
```

Comment:
Ideally we’d only do the sandbox check if these other two checks pass?

Reply:
Thanks for catching, will fix.

Thanks for reviewing!